### DISCUSSION TITLE
Got a frame in a state where it is unexpected error while using CAN-isotp

### DISCUSSION THREAD

**Discussion by aravinth-kalai** [zephyrproject-rtos/zephyr#66479] [Q&A]
Hello everyone, 

Currently I'm working on CAN - isotp, if is send the diagnostics comment through a script , I'm getting this **"Got a frame in a state where it is unexpected"** error from zephyr stack .

Also in my diagnostic script I'm getting 
**WARNING:root:IsoTp error happened : FlowControlTimeoutError - Reception of FlowControl timed out. Stopping transmission
WARNING:isotp:Reception of FlowControl timed out. Stopping transmission** error.

May I know what could be the possible reasons for this error so that I can debug this issue, I'm not sure where to start debug this error

zephyr version : 3.3.0
microcontroller: lpc55s16

Please help me to fix this issue,

Thanks in advance !

**Comment by DerekSnell** [comment]
perhaps @henrikbrixandersen has some insight?

**Comment by henrikbrixandersen** [comment]
@aravinth-kalai Please see if this is reproducible with a more recent version of Zephyr, preferably current `main` branch. Also, please include full verbatim output and, if possible, sample code/code snippets.

**Comment by DerekSnell** [comment]
Closing due to lack of details, and no update

**Reply by aravinth-kalai** [reply]
@henrikbrixandersen Thanks for your comments and sorry for delayed reply, this issue is solved after updating the zephyr 3.3.4 version

**Reply by aravinth-kalai** [reply]
@henrikbrixandersen After adding few functions in my code again this error is occurring, may I know what is the root cause of this issue so that I can solve it, please note if I put some print statements inside the isotp.c  also some time it is working, This issue is uncertain and occurs if we touch the code.
